doc/specs/stdlib_linalg.md

```diff
@@ -979,6 +979,81 @@ This subroutine computes the internal working space requirements for the QR fact
 {!example/linalg/example_qr_space.f90!}
 ```
 
+## `schur` - Compute the Schur decomposition of a matrix
+
+### Status
+
+Experimental
+
+### Description
+
+This subroutine computes the Schur decomposition of a `real` or `complex` matrix: \( A = Z T Z^H \), where \( Z \) is unitary (orthonormal) and \( T \) is upper-triangular (for complex matrices) or quasi-upper-triangular (for real matrices, with possible \( 2 \times 2 \) blocks on the diagonal). Matrix \( A \) has size `[n,n]`.
+
+The results are returned in output matrices \( T \) and \( Z \). Matrix \( T \) is the Schur form, and matrix \( Z \) is the unitary transformation matrix such that \( A = Z T Z^H \). If requested, the eigenvalues of \( T \) can also be returned as a `complex` array of size `[n]`.
+
+### Syntax
+
+`call ` [[stdlib_linalg(module):schur(interface)]] `(a, t, z, [, eigvals] [, overwrite_a] [, storage] [, err])`
+
+### Arguments
+
+- `a`: Shall be a rank-2 `real` or `complex` array containing the matrix to be decomposed. It is an `intent(inout)` argument if `overwrite_a = .true.`; otherwise, it is an `intent(in)` argument.
+
+- `t`: Shall be a rank-2 array of the same kind as `a`, containing the Schur form \( T \) of the matrix. It is an `intent(out)` argument and should have a shape of `[n,n]`.
+
+- `z`: Shall be a rank-2 array of the same kind as `a`, containing the unitary matrix \( Z \). It is an `intent(out)` argument and is optional. If provided, it should have the shape `[n,n]`.
+
+- `eigvals` (optional): Shall be a rank-1 `complex` array containing the eigenvalues of \( A \) (the diagonal elements of \( T \)). The array must be of size `[n]`. If not provided, the eigenvalues are not returned.
```

Comment:
Is it always `complex`? Or should it be of the same kind as `a` and `t`?

Reply:
It is always `complex` and same as we do for `eigs` and `eigvals`. The Python libraries are able to [fall back to `real`](https://github.com/numpy/numpy/blob/2f7fe64b8b6d7591dd208942f1cc74473d5db4cb/numpy/linalg/_linalg.py#L1226-L1231) if all the imaginary parts are `0.0`, but we cannot do that in Fortran because the type is fixed, not dynamic, i.e. NumPy: 

```
    if not isComplexType(t):
        if all(w.imag == 0):
            w = w.real # convert eigs to real
            result_t = _realType(result_t)
        else:
            result_t = _complexType(result_t)
```

Reply:
Sorry: I overlooked that we had implemented such a feature for `eig`. I ported the same feature here, see [daf7a5e](https://github.com/fortran-lang/stdlib/pull/892/commits/daf7a5e23b39ea6b521d63ba0029af752619a35d)